kie-server-parent/kie-server-controller/kie-server-controller-rest/src/main/java/org/kie/server/controller/rest/RestSpecManagementServiceImpl.java

```diff
@@ -69,7 +75,10 @@ public Response saveContainerSpec(@Context HttpHeaders headers,
 
             specManagementService.saveContainerSpec(serverTemplateId, containerSpec);
             logger.debug("Returning response for save container spec request for server template with id '{}': CREATED", serverTemplateId);
-            return createCorrectVariant("", headers, Response.Status.CREATED);
+            String response = marshal(contentType,
+                                      new ServiceResponse(KieServiceResponse.ResponseType.SUCCESS,
+                                                          "KIE container " + containerId + " successfully deployed"));
+            return createCreatedVariant(response, headers, uriInfo.getAbsolutePathBuilder());
```

Comment:
@sterobin I disagree with this statement that every api call should return human-friendly response. Frankly speaking I don't see any need for that as the API docs are meant to provide this human-friendly details that can be easily mapped to response code and potential data returned. Moreover adding additional payload to all request would
- increase time required to receive response as the payload must be marshalled (on both ends)
- increase network use as the number of bytes to be sent will increase
- will in most of the cases be ignored anyway as that is never parsed to present on UI especially when international end users are involved - why would you show English text to Spanish speaking people and since that is technical anyway what's the point anyway? Instead application developers should based their feedback message to UI based on the API docs and response code

You also mentioned that our competitors do this could you please provide examples for it?

Reply:
@mswiderski, you make good points, especially about translation and international considerations. Regarding other APIs, Oracle, HelpSpot, Twitter, Facebook, and other larger organizations are the ones I looked at that do a good job with responses and response messages. Requests typically have a response of some kind, if only repeating back to the user the asset data related to the request, except for delete requests, which typically don't have a response. However, to your point, these aren't our competitors, so if I look at Signavio or the [Camunda API](https://docs.camunda.org/manual/7.7/reference/rest/execution/) for example, which are more our competitors, they follow a method more like what you're saying, but they too are inconsistent.

So I can roll with your logic Maciej, but the only problem I still have is that there are fewer cases of no response messages in the KIE Server REST API than there are with response messages. I was just trying to help us to be consistent in our responses, and I applied the same logic to the controller API for this PR. So imo, we need to choose one route or the other. If we want to go no response messages for reasons you've explained, and just rely on codes, then we need to do this universally. Many KIE Server API requests say "X thing successfully ___". I've just tried to fill in the gaps, but if we don't want to, then those messages should likewise be removed. When we're inconsistent, it looks like the ones without responses are a bug.

So for this PR, @cristianonicolai, I leave it to you whether you want response messages or not. But we need to be consistent.

And @mswiderski, just a reminder that [BXMSDOC-3368](https://issues.jboss.org/browse/BXMSDOC-3368) assigned to you and Tiho is the KIE Server counterpart to this PR, for applying improved API endpoint data to the KIE Serer REST API swagger source. You can likewise choose to either add response messages where I've indicated, or remove response messages where they currently exist, for consistency.